### PR TITLE
Refactor solutions package

### PR DESCRIPTION
**Pull request by weihesdlegend**
## Description
Clearly describe changes in your PR, new feature or bug fixes

## Root Cause
Solutions package has a lot of redundant code and complicated data structures. For example, the physical meanings for `multi-slot-solution` and `slot-solution` were not clear. This also prepares us for new feature development.

## Solution
Refactor a tons of things. I am leaving the POST planning API commented out, will do refactor in another PR. We are not using it anyway.

## Testing
Performed manual integration tests, deleted tests no longer applicable.

### CODE REVIEW

File: solution/solver.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+const (
+	ValidSolutionFound      = 200
+	InvalidRequestLocation  = 400
+	ReqTagInvalid           = 400
+	CatPlaceIterInitFailure = 404
+	NoValidSolution         = 404
+)
+
+type PlanningSolutions struct {
+	SlotSolutions []PlanningSolution
```
Comment (xtpitt) [inline solution/solver.go:31]: rename slotsolution

**Review by xtpitt — COMMENTED** [review]
See inline comment.

File: solution/solver.go (hunk 1 above)
Comment (weihesdlegend) [reply solution/solver.go:31]: Removed since it's not used anywhere.

File: solution/solver.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	TravelTimes   []uint
```
[the remark this answers is not in the document]
Comment (weihesdlegend) [reply solution/solver.go:32]: Done

File: solution/solver.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+			sb.WriteString("v")
+		}
+	}
+	redisRequests[0] = GenerateSlotSolutionRedisRequest(req.Location, sb.String(), ToTimeSlots(req.Slots), req.SearchRadius, req.Weekday)
+
+	slotSolutionCacheResponses := redisCli.GetMultiSlotSolutions(context, redisRequests)
+
+	slotSolutionRedisKeys := make([]string, len(redisRequests))
+
+	solution := slotSolutionCacheResponses[0]
```
[the remark this answers is not in the document]
Comment (weihesdlegend) [reply solution/solver.go:129]: Done

File: solution/solver.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+		switch cat {
+		case POI.PlaceCategoryEatery:
+			sb.WriteString("e")
+		case POI.PlaceCategoryVisit:
+			sb.WriteString("v")
+		}
+	}
+	redisRequests[0] = GenerateSlotSolutionRedisRequest(req.Location, sb.String(), ToTimeSlots(req.Slots), req.SearchRadius, req.Weekday)
+
+	slotSolutionCacheResponses := redisCli.GetMultiSlotSolutions(context, redisRequests)
```
[the remark this answers is not in the document]
Comment (weihesdlegend) [reply solution/solver.go:125]: Added todo to fix later

**Review by xtpitt — APPROVED** [review]
GJ

**Pull request merged**